src/structures/tickets.rs

```diff
@@ -66,27 +70,70 @@ macro_rules! impl_ticket_try_froms {
 
                 let hierarchy = Hierarchy::try_from(TpmHandle::try_from(tss_ticket.hierarchy)?)?;
 
-                Ok($ticket_type { hierarchy, digest })
+                Ok($ticket_type {
+                    tag,
+                    hierarchy,
+                    digest,
+                })
             }
         }
     };
 }
 
 pub trait Ticket {
-    const TAG: StructureTag;
+    const POSSIBLE_TAGS: &'static [StructureTag];
+    fn tag(&self) -> StructureTag {
```

Comment:
Nice! I like this.

Reply:
Thanks. I spent a long time coming up with this approach (longer than I'd like to admit 😀).